CRM/Contribute/Form/ContributionBase.php

```diff
@@ -893,7 +893,7 @@ public function buildComponentForm($id, $form) {
           }
         }
 
-        $form->assign('fieldSetTitle', ts('Organization Details'));
+        $form->assign('fieldSetTitle', CRM_Core_BAO_UFGroup::getTitle($form->_values['onbehalf_profile_id']));
```

Comment:
It seems that ts() on `Organization Details` was displaying the converted string for each language. This seems to be working fine with the intention of the issue. But the organization title will only be shown in English after this change. Any reason of not assigning the title to ts() before assigning to `$form` ?